device/transport/mqtt/src/mqtt.ts

```diff
@@ -277,6 +277,12 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
               topic += '/';
             }
 
+            // This will not catch all messages that exceed IoT Hub limits because properties contribute the size as well.
+            if ((message?.data?.length ?? 0) > 256 * 1024) {
```

Comment:
>message?.data?.length  [](http://example.com/codeflow?start=17&length=22)

`message.data` is typed as `any`, but not all objects have a `length` attribute. JSON objects don't,, but I can't see anywhere below here that would serialize an object. 

Regardless, `publish()` in mqtt-packet has code to handle Buffer objects when calculating packet length.  You may or may not want to replicate that code here.

Reply:
message.data must be a string or Buffer according to the MQTT.js typing. I think whoever wrote the types here made a mistake.